### ISSUE TITLE
`Before(Test)` hook is not called but `After(Test)` hook is for a `Skipped` test

### ISSUE THREAD

**Issue by c-schembri** [thomhurst/TUnit#817]
In the below code snippet, the `TearDown` method is executed but the `SetUp` method is not executed--possibly causing exceptions to be thrown. I don't know what the solution is, because it seems to be a design decision. But I believe that if you want to execute the `After` hook then the `Before` hook should be called too and vice versa. I don't think you'd ever want a situation when only one of these hooks are called.

```
public class DummyHooks
{
    public string? _dummy;
    
    [Before(Test)]
    public async Task SetUp()
    {
        _dummy = "Before"; // This line never executes.
    }
    
    [After(Test)]
    public async Task TearDown()
    {
        _dummy = "After"; // Exception thrown here.
    }
}

public class SkippedTestWithHooksTest : DummyHooks
{
    [Test, Skip("Skipped")]
    public async Task Test()
    {
    }
}
```

**Comment by thomhurst** [comment]
Yeah this isn't meant to be called. Thanks, I'll take a look!

**Comment by thomhurst** [comment]
Should be resolved in 0.1.997
